### PR TITLE
Use ClientInterface instead of HttpClient

### PR DESCRIPTION
**Pull request by glaubinix**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | yes
| Deprecations?   | no
| Related tickets | addresses one more item of https://github.com/php-http/HttplugBundle/pull/439#issuecomment-1820427669
| Documentation   | if this is a new feature, link to pull request in https://github.com/php-http/documentation that adds relevant documentation
| License         | MIT

#### What's in this PR?

Instead of using the deprecated `HttpClient` interface, this now uses `ClientInterface` everywhere

#### Why?

The old interface is deprecated

#### Checklist

- [x] Updated CHANGELOG.md to describe BC breaks / deprecations | new feature | bugfix
- [ ] Documentation pull request created (if not simply a bugfix)

#### To Do

- [ ] I am not sure if keeping around both `httplug.client.default` and `httplug.psr18_client.default` is OK or if this will cause any issues? If we can keep them around, then it might be easier for people to update the library without running into issues.

### CODE REVIEW

File: tests/Unit/Collector/ProfileCustomClientTest.php [hunk 1]
```diff
@@ -240,6 +240,6 @@ public function testOnRejected(): void
     }
 }
 
-interface ClientInterface extends HttpClient, HttpAsyncClient
+interface CustomClientInterface extends ClientInterface, HttpAsyncClient
```
Comment (dbu) [inline tests/Unit/Collector/ProfileCustomClientTest.php]:
```suggestion
interface CombinedClientInterface extends ClientInterface, HttpAsyncClient
```

**Review by dbu — APPROVED** [review]
thanks a lot!

i have one small nitpick on the naming in the test.

**Review by dbu — APPROVED** [review]
thanks!

we should update a few things on https://docs.php-http.org/en/latest/integrations/symfony-bundle.html and https://docs.php-http.org/en/latest/integrations/symfony-full-configuration.html (sources are at https://github.com/php-http/documentation/tree/main/integrations)

**Comment by glaubinix** [comment]
Yes, that was on my list. How do you generally handle cases that are different in versions, e.g. removed config keys?

**Comment by dbu** [comment]
generally it would be best if we had a separate documentation for each major version, but the doc is for all the various components together so no meaningful versioning is possible.

i would just remove obsolete things, and have `..versionadded:: 2.0` notes for new things (see for example components/client-common.rst)

mostly people will read the doc when setting things up. for BC breaks the changelog of the corresponding repository is the right place.

**Pull request merged**